README.Rmd

```diff
@@ -0,0 +1,104 @@
+---
+output: github_document
+---
+
+<!-- README.md is generated from README.Rmd. Please edit that file -->
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.path = "man/figures/README-",
+  out.width = "100%"
+)
+```
+
+# grtsdb
+
+<!-- badges: start -->
+[![Project Status: Active – The project has reached a stable, usable state and is being actively developed.](https://www.repostatus.org/badges/latest/active.svg)](https://www.repostatus.org/#active)
+[![Lifecycle: maturing](https://img.shields.io/badge/lifecycle-maturing-blue.svg)](https://www.tidyverse.org/lifecycle/#maturing)
+![GitHub](https://img.shields.io/github/license/inbo/grtsdb)
+[![R build status](https://github.com/inbo/grtsdb/workflows/R-CMD-check/badge.svg)](https://github.com/inbo/grtsdb/actions)
+[![Codecov test coverage](https://codecov.io/gh/inbo/grtsdb/branch/master/graph/badge.svg)](https://codecov.io/gh/inbo/grtsdb?branch=master)
+![GitHub code size in bytes](https://img.shields.io/github/languages/code-size/inbo/grtsdb.svg)
+![GitHub repo size](https://img.shields.io/github/repo-size/inbo/grtsdb.svg)
+<!-- badges: end -->
+
+The goal of `grtsdb` is to create as statially balanced sample based on the 'Generalised Random Tesselation Stratified' strategy.
+We store the base schema in an SQLite database.
+Store this database to make the sampling reproducible.
+Sampling the same database with the same parameters yields a stable sample.
+
+## Installation
+
+You can install the development version from [GitHub](https://github.com/) with:
+
+``` r
+# install.packages("remotes")
+remotes::install_github("inbo/grtsdb")
+```
+
+## Example
+
+This is a basic example.
+
+Connect to a database.
+
+```{r}
+tmp_copy <- tempfile(pattern = "grts", fileext = ".sqlite")
+if (system.file("grts.sqlite", package = "grtsdb") != "") {
+  file.copy(system.file("grts.sqlite", package = "grtsdb"), tmp_copy)
+}
+library(grtsdb)
+db <- connect_db(tmp_copy)
+```
+
+To extract a sample, you'll need to specify the bounding box in projected
+coordinates and the size of the grid cells.
+
+```{r example}
+bbox <- rbind(
+  c(0, 32),
+  c(0, 32)
+)
+extract_sample(grtsdb = db, samplesize = 10, bbox = bbox, cellsize = 1)
```

Comment:
No need to use `set.seed()`?

Reply:
Unfortunately, you can't set a seed with SQLite. Therefore you must archive the SQLite database to reproduce it later.

Reply:
Probably not in scope directly, but https://arrow.apache.org/docs/r/ looks like a very promising alternative for SQLite along with the `parquet` data storage type. The way GRTS data are stored is I think ideal for these tools. Learned about it at the rstudio::global conference last week.

Reply:
Downside of arrow is that you need to install extra dependencies. SQlite works out of the box on different OS.

Reply:
According to https://arrow.apache.org/docs/r/#installation, the released versions of the package require no additional system dependencies. I would consider this only if speed and / or size on disk becomes an issue. This will not be the case for Belgium, but it may well become problematic if you want to use grtsdb for - say - a master sample in Europe at high resolution.

Reply:
Flanders at 10 m resolution is about 50 GB. We need one extra level for Europe at 100 m level.

Reply:
That sounds doable :+1:

Reply:
FYI https://tech.marksblogg.com/billion-nyc-taxi-rides-sqlite-parquet-hdfs.html

Reply:
> Unfortunately, you can't set a seed with SQLite. Therefore you must archive the SQLite database to reproduce it later.

Just a follow-up on the above comments, because I've been reading about [duckdb]((https://duckdb.org/) as an alternative to SQLite recently and wanted to share some insights that could interest you for grtsdb:

- It has no external dependencies and is extensively tested across many platforms (https://duckdb.org/docs/why_duckdb)
- allows a seed in the sample statement (https://duckdb.org/docs/sql/samples)
- making SQLite queries fast, requires indexes to be built (which take a lot of disk space) - duckdb is fast out-of-the-box due to it being columnar-vectorized 
- can send SQL statements directly on top of parquet datasets (https://duckdb.org/2021/06/25/querying-parquet.html) greatly reducing required storage while still very fast to query

I've done no tests with the package yet, but from what I've read it feels like it can be used as a drop-in replacement for SQLite with these extra benefits.